scripts/postgres/schema-v1-postgresql.sql

```diff
@@ -0,0 +1,111 @@
+-- set search_path to polaris_schema;
+
+CREATE TABLE IF NOT EXISTS entities (
+    catalog_id BIGINT NOT NULL,
+    id BIGINT NOT NULL,
+    parent_id BIGINT NOT NULL,
+    name VARCHAR(126) NOT NULL,
+    entity_version INT NOT NULL,
+    type_code INT NOT NULL,
+    sub_type_code INT NOT NULL,
+    create_timestamp BIGINT NOT NULL,
+    drop_timestamp BIGINT NOT NULL,
+    purge_timestamp BIGINT NOT NULL,
+    to_purge_timestamp BIGINT NOT NULL,
+    last_update_timestamp BIGINT NOT NULL,
+    properties JSONB not null default '{}'::JSONB,
+    internal_properties JSONB not null default '{}'::JSONB,
+    grant_records_version INT NOT NULL,
+    PRIMARY KEY (catalog_id, id),
+    CONSTRAINT constraint_name UNIQUE (catalog_id, parent_id, type_code, name)
+);
+
+COMMENT ON TABLE entities IS 'all the entities';
+
+COMMENT ON COLUMN entities.catalog_id IS 'catalog id';
+COMMENT ON COLUMN entities.id IS 'entity id';
+COMMENT ON COLUMN entities.parent_id IS 'entity id of parent';
+COMMENT ON COLUMN entities.name IS 'entity name';
+COMMENT ON COLUMN entities.entity_version IS 'version of the entity';
+COMMENT ON COLUMN entities.type_code IS 'type code';
+COMMENT ON COLUMN entities.sub_type_code IS 'sub type of entity';
+COMMENT ON COLUMN entities.create_timestamp IS 'creation time of entity';
+COMMENT ON COLUMN entities.drop_timestamp IS 'time of drop of entity';
+COMMENT ON COLUMN entities.purge_timestamp IS 'time to start purging entity';
+COMMENT ON COLUMN entities.last_update_timestamp IS 'last time the entity is touched';
+COMMENT ON COLUMN entities.properties IS 'entities properties json';
+COMMENT ON COLUMN entities.internal_properties IS 'entities internal properties json';
+COMMENT ON COLUMN entities.grant_records_version IS 'grant record version of the entity';
+
+CREATE TABLE IF NOT EXISTS grant_records (
+    securable_catalog_id BIGINT NOT NULL,
+    securable_id BIGINT NOT NULL,
+    grantee_catalog_id BIGINT NOT NULL,
+    grantee_id BIGINT NOT NULL,
+    privilege_code INTEGER,
+    PRIMARY KEY (securable_catalog_id, securable_id, grantee_catalog_id, grantee_id, privilege_code)
+);
+
+CREATE INDEX IF NOT EXISTS idx_grant_records ON grant_records (securable_catalog_id, securable_id, grantee_catalog_id, grantee_id);
+
+COMMENT ON TABLE grant_records IS 'grant records';
+
+COMMENT ON COLUMN grant_records.securable_catalog_id IS 'catalog id of the securable';
+COMMENT ON COLUMN grant_records.securable_id IS 'entity id of the securable';
+COMMENT ON COLUMN grant_records.grantee_catalog_id IS 'catalog id of the grantee';
+COMMENT ON COLUMN grant_records.grantee_id IS 'id of the grantee';
+COMMENT ON COLUMN grant_records.privilege_code IS 'privilege id';
+
+
+CREATE TABLE IF NOT EXISTS principal_secrets (
+    principal_id BIGINT NOT NULL,
+    principal_client_id VARCHAR(500) NOT NULL,
+    main_secret_hash VARCHAR(500) NOT NULL,
+    secondary_secret_hash VARCHAR(500) NOT NULL,
+    secret_salt VARCHAR(500) NOT NULL,
```

Comment:
Why 500 chars?

Reply:
I tried 100 it failed so i set it a large value, let me check precisely what should be suff.

Reply:
changed this to 255 looks like we do 32 byte byte array to hexa-decimal string which ideally should not require this, changing this to the eclipe link for now.